src/platform/user/profile/actions/mhv.js

```diff
@@ -67,8 +68,15 @@ export function upgradeMHVAccount() {
 
 export function createAndUpgradeMHVAccount() {
   return async dispatch => {
-    const accountCreationSuccess = await dispatch(createMHVAccount());
-    if (accountCreationSuccess) return dispatch(upgradeMHVAccount());
+    const accountCreationResult = await dispatch(createMHVAccount());
+
+    const accountLevel = get(
+      'data.attributes.accountLevel',
+      accountCreationResult,
+    );
+
+    if (accountLevel) return dispatch(upgradeMHVAccount());
```

Comment:
Local testing seems to show that this block didn't get run before, since we weren't returning anything at the end of `createMHVAccount`...